### PR TITLE
Refactor some functions to return pointers

### LINKED ISSUE #95: Use pointers for better error handling
When returning a struct and error from a function, it’s better to return a pointer to a struct and error so if an error occurs the result is nil. This will prevent downstream code to process an erroneous empty result if the error is not checked. Example: `LatestBlockId` function.

### PR DESCRIPTION
**Pull request by nanuuki**
Resolve #95

### CODE REVIEW

File: keys/google/google_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"context"
+	"testing"
+
+	"github.com/flow-hydraulics/flow-wallet-api/configs"
+	"github.com/flow-hydraulics/flow-wallet-api/keys"
+)
+
+func TestGenerate(t *testing.T) {
+	opts := &configs.Options{EnvFilePath: "../../.env.test"}
+	testCfg, err := configs.ParseConfig(opts)
```
Comment (latenssi) [inline keys/google/google_test.go:13]: I would manually set `testCfg.DatabaseDSN` to something safe here, just in case.

**Review by latenssi — APPROVED** [review]
One minor note, other than that LGTM 👍

**Comment by nanuuki** [comment]
Note; there are currently leaking goroutines when running the whole test suite with Google KMS keys. 

Related issues are;
* #167 
* onflow/flow-go-sdk/issues/194

File: keys/google/google_test.go (hunk 1 above)
Comment (nanuuki) [reply keys/google/google_test.go:14]: The tests shouldn't touch the database here, but better safe than sorry :)

**Pull request merged**